roles/postgres-infra/templates/postgres-sfset.yaml.j2

```diff
@@ -0,0 +1,42 @@
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+ name: postgres{{cluster_number}}
+ namespace: '{{ operator_namespace }}'
+spec:
+ serviceName: "postgres{{cluster_number}}"
+ replicas: {{ postgres.cluster_size }}
+ template:
+   metadata:
+     labels:
+       role: postgres
+   spec:
+     terminationGracePeriodSeconds: 10
+     containers:
+       - name: postgres
+         image: postgres:10.4
+         envFrom:
+           - configMapRef:
+               name: postgres-config
+         ports:
+{% if postgres.port is not defined %}
+           - containerPort: 5432
+{% endif %}
+{% if postgres.port is defined %}
+           - containerPort: {{ postgres.port }}
+{% endif %}
+         volumeMounts:
+           - name: postgres-persistent-storage
+             mountPath: /var/lib/postgresql/data
+ volumeClaimTemplates:
+   - metadata:
+       name: postgres-persistent-storage
+{% if postgres.storageclass is defined %}
+       annotations:
+         volume.beta.kubernetes.io/storage-class: {{ postgres.storageclass }}
+{% endif %}
+     spec:
+       accessModes: [ "ReadWriteOnce" ]
+       resources:
+         requests:
+           storage: {{ postgres.storagesize }}
```

Comment:
I would modify the CR and template logic so that a `storageClass` is not a requirement. You could use an `{% if %}` statement in the template based on the CR vars and use that to either define the `volumeClaimTemplate` here or or otherwise just define a `volumes` section in the container spec to set postgres-persistent-storage to an `emptyDir`

Reply:
so a storageclass isn't a requirement in cr, it's commented out in the docs. About emptyDir, I don't see us/anyone else deploying postgres like that. However if they want to, they can simply use a sfset/service definition and do it. I'd so thats outside the scope of this pr, and the work we'd want to do, at least from my pov. However if you think its a need, we can deal with it in a subsequent pr.